version: '3.8'
services:
  redis:
    image: redis:7-alpine
    container_name: redis_test_container
    ports:
      - "${REDIS_PORT:-6379}:6379"
    volumes:
      - redis_data:/data
    networks:
      - postgres
    restart: unless-stopped
  postgis:
    image: postgis/postgis:16-3.4
    container_name: postgis_test_container
    environment:
      POSTGRES_DB: ${DB_NAME:-allpass_test}
      POSTGRES_USER: ${DB_USER:-zoe}
      POSTGRES_PASSWORD: ${DB_PASSWORD:-aipe-tester}
    volumes:
      - postgis_test:/var/lib/postgresql/data
      - ./init:/docker-entrypoint-initdb.d
    ports:
      - "${DB_HOST_PORT:-5432}:5432"
    networks:
      - postgres
    restart: unless-stopped
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U ${DB_USER:-zoe} -d ${DB_NAME:-allpass_test}" ]
      interval: 5s
      timeout: 5s
      retries: 5

volumes:
  redis_data:
    name: my_redis_test
  postgis_test:
    name: my_postgis_test

networks:
  postgres: 
    # external: True
